astropy/uncertainty/builtin_distrs.py

```diff
@@ -0,0 +1,192 @@
+    dcls : class
+        The class to use to create this distribution.  Typically a
+        `Distribution` subclass.
+
+    Remaining keywords are passed into the constructor of the ``dcls``
+
+    Returns
+    -------
+    distr : ``dcls``, usually `Distribution`
+        The sampled poisson distribution.
+    """
+    # we convert to arrays because np.random.poisson has trouble with quantities
+    has_unit = False
+    if hasattr(center, 'unit'):
+        has_unit = True
+        poissonarr = np.asanyarray(center.value)
+    else:
+        poissonarr = np.asanyarray(center)
+    randshape = poissonarr.shape + (n_samples,)
+
+    samples = np.random.poisson(poissonarr[..., np.newaxis], randshape)
+    if has_unit:
+        # re-attach the unit
+        samples = samples * center.unit
+
+    distr = dcls(samples, **kwargs)
+    distr.distr_std = center**0.5
+    return distr
+
+
+def uniform(lower, upper, n_samples=1000, dcls=Distribution, **kwargs):
+    """
+    Create a Uniform Distribution.
+
+    Parameters
+    ----------
+    lower : array-like
+        The lower edge of this distribution. If a `~astropy.units.Quantity`, the
+        distribution will have the same units as ``lower``.
+    upper : `~astropy.units.Quantity`
+        The upper edge of this distribution. Must match shape and if a
+        `~astropy.units.Quantity` must have compatible units with ``lower``.
+    n_samples : int
+        The number of Monte Carlo samples to use with this distribution
+    dcls : class
+        The class to use to create this distribution.  Typically a
+        `Distribution` subclass.
+
+    Remaining keywords are passed into the constructor of the ``dcls``
+
+    Returns
+    -------
+    distr : ``dcls``, usually `Distribution`
+        The sampled uniform distribution.
+    """
+    lhasu = hasattr(lower, 'unit')
+    uhasu = hasattr(upper, 'unit')
+    unit = None
+    if lhasu and uhasu:
+        if lower.unit != upper.unit:
+            upper = upper.to(lower.unit)
+        unit = lower.unit
+
+        lowerarr = np.asanyarray(lower.value)
+        upperarr = np.asanyarray(upper.value)
+    elif not lhasu and not uhasu:
+        lowerarr = np.asanyarray(lower)
+        upperarr = np.asanyarray(upper)
+    else:
+        raise UnitsError('lower and upper must have consistent (or no) '
+                         'units in uniform.')
+
+    if lowerarr.shape != upperarr.shape:
+        raise ValueError('lower and upper must have consistent shapes in '
+                         'uniform.')
+
+    newshape = lowerarr.shape + (n_samples,)
+    samples = np.random.uniform(lowerarr[..., np.newaxis],
+                                upperarr[..., np.newaxis], newshape)
+    if unit is not None:
+        samples = samples * unit
+
+    return dcls(samples, **kwargs)
+
+
+def uniform_center_width(center, width, **kwargs):
+    """
+    Create a uniform distribution from lower/upper bounds (instead of center
```

Comment:
For a normal distribution, we have one function with three options (sigma, fwhm, var). Why not do the same for uniform and have keyword-only arguments, so e.g:

```
def uniform(*, center=None, width=None, min=None, max=None)
```

I don't feel strongly about this, but it could be a nice way to always be explicit.